torchtune/modules/attention.py

```diff
@@ -164,6 +169,7 @@ def setup_cache(
                 head_dim=self.head_dim,
                 dtype=dtype,
             )
+            self.cache_enabled = True
```

Comment:
So I'm setting to True here to preserve the original behaviour of `setup_caches()` -> all forwards passes use cache, and it means the new context generators are the recommended default we showcase in the recipes, but you can still do it the old way if you want.

This may be slightly unintuitive if you want to use the persistent context manager, i.e. setup caches somewhere early in your recipe, and then have some forward passes *before* you want to use caches.

If we set this to False by default it means the expected API for using KV-caches is through the context managers i.e. BC break

Reply:
I actually think using context managers as the de-facto KV-cache API is sensible but open to thoughts here.